main.js

```diff
@@ -125,19 +137,27 @@ function persistOnPageLoad() {
   }
 }
 
-
 function activeSearchFilter() {
   var titleArray = document.querySelectorAll('.idea-title');
   var bodyArray = document.querySelectorAll('.idea-body');
   var titleCharacters;
   var bodyCharacters;
-  for(i = 0; i < titleArray.length; i++) {
+  for (i = 0; i < titleArray.length; i++) {
     titleCharacters = titleArray[i].innerText.toLowerCase();
     bodyCharacters = bodyArray[i].innerText.toLowerCase();
-      if(titleCharacters.includes(event.target.value.toLowerCase()) || bodyCharacters.includes(event.target.value.toLowerCase())) {
-        titleArray[i].parentNode.parentNode.classList.remove('hidden');
-      } else {
-          titleArray[i].parentNode.parentNode.classList.add('hidden');
-        }
+    if (titleCharacters.includes(searchBox.value.toLowerCase()) || bodyCharacters.includes(searchBox.value.toLowerCase())) {
+      titleArray[i].parentNode.parentNode.classList.remove('hidden');
+    } else {
+      titleArray[i].parentNode.parentNode.classList.add('hidden');
+    }
   }
 }
+
+function showStarredIdeas() {
+  if (showStarredIdeasButton.innerText === `Show Starred Ideas`) {
+    showStarredIdeasButton.innerText = `Show All Ideas`;
+  } else {
+    showStarredIdeasButton.innerText = `Show Starred Ideas`;
+  }
+  displayCards();
+}
```

Comment:
When you click on the Show Starred Ideas button, the showStarredIdeas function will be invoked. If the innerText of the showStarredIdeasButton is `Show Starred Ideas`, then we want to change/reassign it to `Show All Ideas` and vice versa. The displayCards function is invoked.